lib/streamlit/navigation/page.py

```diff
@@ -41,11 +41,13 @@ def Page(
     ``st.navigation`` to declare a page in your app.
 
     When a user navigates to a page, ``st.navigation`` returns the selected
-    ``StreamlitPage`` object. Call ``StreamlitPage.run()`` on the returned page
-    to execute the page. You can only run the page returned by
+    ``StreamlitPage`` object. Call ``.run()`` on the returned ``StreamlitPage``
+    object to execute the page. You can only run the page returned by
     ``st.navigation``, and you can only run it once per app rerun.
 
-    A page can be defined by a Python file or ``Callable``.
+    A page can be defined by a Python file or ``Callable``. Since a page is
+    executed from within the entrypoint file, page source code should not
+    include ``if __name__ == "__main__":`` blocks.
```

Comment:
What about something like

> Python files used as StreamlitPage source will have `__name__ == "__page__"`. Only the entrypoint file has `__name__ == "__main__"` to adhere to Python convention.

Reply:
Yeah, happy to give more complete information.

Reply:
I just did a quick test to confirm what happens with a defined function (it get "__main__" if it was defined in the entrypoint file and the name of its module if imported from elsewhere (even if that elsewhere was another page).

> Python files used as a `StreamlitPage` source will have `__name__ == "__page__"`. Functions used as `StreamlitPage` source will have `__name__` corresponding to the module they were imported from. Only the entrypoint file and functions defined within the entrypoint file have `__name__ == "__main__"` to adhere to Python convention.

Reply:
Yep that version sounds good! Thanks!!